packages/vite/src/node/build.ts

```diff
@@ -895,43 +888,74 @@ export function onRollupWarning(
         }
       }
 
-      if (
-        warning.plugin === 'rollup-plugin-dynamic-import-variables' &&
-        dynamicImportWarningIgnoreList.some((msg) =>
-          // @ts-expect-error warning is RollupLog
-          warning.message.includes(msg),
-        )
-      ) {
-        return
-      }
+      if (logLeveling === 'warn') {
+        if (
+          logging.plugin === 'rollup-plugin-dynamic-import-variables' &&
+          dynamicImportWarningIgnoreList.some((msg) =>
+            logging.message.includes(msg),
+          )
+        ) {
+          return
+        }
 
-      if (warningIgnoreList.includes(warning.code!)) {
-        return
+        if (warningIgnoreList.includes(logging.code!)) {
+          return
+        }
       }
 
-      if (warning.code === 'PLUGIN_WARNING') {
-        config.logger.warn(
-          `${colors.bold(
-            colors.yellow(`[plugin:${warning.plugin}]`),
-          )} ${colors.yellow(warning.message)}`,
-        )
-        return
+      if (logging.code?.startsWith('PLUGIN_')) {
+        switch (logLeveling) {
+          case 'info':
+            config.logger.info(
+              `${colors.bold(
+                colors.green(`[plugin:${logging.plugin}]`),
+              )} ${colors.green(logging.message)}`,
+            )
+            return
+
+          case 'warn':
+            config.logger.warn(
+              `${colors.bold(
+                colors.yellow(`[plugin:${logging.plugin}]`),
+              )} ${colors.yellow(logging.message)}`,
+            )
+            return
+
+          case 'error':
+            config.logger.error(
+              `${colors.bold(
+                colors.red(`[plugin:${logging.plugin}]`),
+              )} ${colors.red(logging.message)}`,
+            )
+            return
+
+          case 'debug':
+            config.logger.info(
+              `${colors.bold(
+                colors.gray(`[plugin:${logging.plugin}]`),
+              )} ${colors.gray(logging.message)}`,
+            )
```

Comment:
We use `debug` function for debug logs in other places. Let's use it here as well.
https://github.com/vitejs/vite/blob/1292ad08a06e2ae8c44307797ba533f64a602759/packages/vite/src/node/utils.ts#L157C17-L157C31

Reply:
@sapphi-red 
Thank you for your review.
I missed vite's debugger implementation. I pushed the commit fixing it to use Vite's debugger.
Please check it.